vic/drivers/shared_image/include/vic_driver_shared_image.h

```diff
@@ -160,56 +160,55 @@ typedef struct {
  * @brief   file structures
  *****************************************************************************/
 typedef struct {
-    FILE *forcing[MAX_FORCE_FILES];   /**< forcing data files */
     FILE *globalparam;  /**< global parameters file */
     FILE *constants;    /**< model constants parameter file */
-    FILE *domain;       /**< domain file */
-    FILE *init_state;   /**< initial model state file */
-    FILE *paramfile;    /**< parameter file */
-    FILE *statefile;    /**< output model state file */
     FILE *logfile;      /**< log file */
 } filep_struct;
 
 /******************************************************************************
  * @brief   This structure stores input and output filenames.
  *****************************************************************************/
 typedef struct {
-    char forcing[MAX_FORCE_FILES][MAXSTRING];    /**< atmospheric forcing data file names */
+    nameid_struct forcing[MAX_FORCE_FILES];    /**< atmospheric forcing data file names */
     char f_path_pfx[MAX_FORCE_FILES][MAXSTRING]; /**< path and prefix for atmospheric forcing data file names */
     char global[MAXSTRING];        /**< global control file name */
-    char domain[MAXSTRING];        /**< domain file name */
+    nameid_struct domain;        /**< domain file name and nc_id*/
     char constants[MAXSTRING];     /**< model constants file name */
-    char params[MAXSTRING];        /**< model parameters file name */
-    char init_state[MAXSTRING];    /**< initial model state file name */
+    nameid_struct params;        /**< model parameters file name and nc_id */
+    nameid_struct init_state;    /**< initial model state file name and nc_id */
     char result_dir[MAXSTRING];    /**< directory where results will be written */
     char statefile[MAXSTRING];     /**< name of file in which to store model state */
     char log_path[MAXSTRING];      /**< Location to write log file to */
 } filenames_struct;
 
-void add_nveg_to_global_domain(char *nc_name, domain_struct *global_domain);
+void add_nveg_to_global_domain(nameid_struct nc_nameid,
```

Comment:
Should pass pointers to structures rather than structures. Passing structures means a copy which is more expensive. If you are concerned that the values are changed you can use `const`s.

So 
``` C
void add_nveg_to_global_domain(nameid_struct *nc_nameid,
```